crates/sui/src/client_commands.rs

```diff
@@ -2247,3 +2314,86 @@ pub async fn request_tokens_from_faucet(
     }
     Ok(())
 }
+
+fn pretty_print_balance(
+    coins: &BTreeMap<String, (Option<SuiCoinMetadata>, Vec<Coin>)>,
+    builder: &mut TableBuilder,
+    with_coins: &bool,
+) {
+    fn print(
+        builder: &mut TableBuilder,
+        coins: Vec<&(Option<SuiCoinMetadata>, Vec<Coin>)>,
+        with_coins: &bool,
+    ) {
+        let mut table_builder = TableBuilder::default();
+        for (metadata, coins) in coins {
+            let (name, symbol, decimals) = if let Some(metadata) = metadata {
+                (
+                    metadata.name.clone(),
+                    metadata.symbol.clone(),
+                    metadata.decimals,
+                )
+            } else {
+                ("unknown".to_string(), "unknown symbol".to_string(), 10u8)
+            };
+
+            if *with_coins {
+                let coin_numbers = if coins.len() > 1 { "coins" } else { "coin" };
+                builder.push_record(vec![format!(
+                    "{}: {} {coin_numbers}, Balance: {:<2.2} {}\n ┌",
+                    name,
+                    coins.len(),
+                    coins.iter().map(|x| x.balance).sum::<u64>() as f64
+                        / 10u64.pow(decimals as u32) as f64,
```

Comment:
I don't think we display balances using decimals like this anywhere else, so I recommend we do the simple thing here and continue ignoring `decimals`, but if we did want to pay attention to it, we shouldn't do it by going via a float, which risks loss of precision issues.

Reply:
I would like that we display the values accordingly though, so I'll think about it.

Reply:
That's fine, but given the additional complexity in how that needs to be handled, and the fact that there are other client commands that represent balances without the decimal, to avoid confusion and make sure that change gets due attention, let's tackle it as a separate change that is applied uniformly.

Reply:
Ok, let's do that.